zio/zngio/scanner.go

```diff
@@ -59,6 +133,16 @@ func (s *zngScanner) Pull() (zbuf.Batch, error) {
 }
 
 func (s *zngScanner) scanUncompressed() ([]*zng.Record, error) {
+	if s.stringFinder != nil {
+		buf := s.reader.uncompressed.Bytes()
+		if s.stringFinder.next(byteconv.UnsafeString(buf)) == -1 {
+			// We know s.reader.uncompressed cannot contain any
+			// records matching s.filter.
+			s.reader.uncompressed = nil
+			// xxx stats
```

Comment:
Another issue here is that the obvious way to update scanner.ScannerStats.BytesRead is to add len(buf) to it, but that will include value message header bytes (the uvarint7 and uvarint containing the type ID and the value length), which aren't included by scanner.scanner or by zngScanner.scanOne.

In other words, scanner.ScannerStats.BytesRead can vary with the filter. That feels wrong, but so does the obvious fix (changing the ZNG spec so that a compressed value mesasge block stores the number of non-header bytes it contains).

Reply:
This is the header you're referring to, correct?
https://github.com/brimsec/zq/blob/a1a817e0a3d2e1781807df51533bd7794d97d8f7/zio/zngio/writer.go#L158

If so, I think it's valid to include that in BytesRead, since it reflects bytes that needed to be read to pull data.

Reply:
No, I'm referring to the header on each value message. It's added here.
https://github.com/brimsec/zq/blob/a1a817e0a3d2e1781807df51533bd7794d97d8f7/zio/zngio/writer.go#L82-L90

Reply:
I am probably missing something, but why can't BytesRead just be derived from a count of bytes Read by the underlying io.Reader (probably by wrapping it with a counter)?

Reply:
> why can't BytesRead just be derived from a count of bytes Read by the underlying io.Reader (probably by wrapping it with a counter)?

It certainly could. The catch is that that would change behavior, though I don't know if anyone cares. (Right now, a given query against a given set of records will produce the same stats regardless of whether those records are stored as Zeek TSV, JSON, TZNG, or ZNG. That would change.)

Reply:
I'd like to move forward with what's here (BytesRead includes value message headers for skipped compressed message value blocks only) and revisit the question for #1010.